### PR TITLE
Feature/small spec and codegen alignments

### PR DESCRIPTION
**Pull request by patrick-tolosa**
#### What does this PR solve?
See commit list in the PR for tasks done. (each commit is isolated)

#### Final checklist
- [X] Implementation is aligned to Spec.
- [X] Tests - Add proper tests to the added code.

### CODE REVIEW

File: src/actions/effect/vectorize.ts [hunk 1]
```diff
@@ -31,7 +31,7 @@ class VectorizeEffectAction extends Action {
    * @param {number} num
```
Comment (strausr) [inline src/actions/effect/vectorize.ts:31]: add string

File: src/actions/effect/vectorize.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ class VectorizeEffectAction extends Action {
...
    * @return {this}
    */
-  despeckleLevel(num: number):this {
+  despeckleLevel(num: number | string):this {
```
Comment (strausr) [inline src/actions/effect/vectorize.ts:44]: add string

File: src/assets/CloudinaryFile.ts [hunk 3]
```diff
@@ -57,6 +57,9 @@ class CloudinaryFile {
    * @return {string} CloudinaryURL
    */
   createCloudinaryURL(transformation?: Transformation): string {
+    if (typeof this.cloudName === 'undefined') {
+      throw 'You must supply a cloudName in either toURL() or whe initializing the asset';
```
Comment (strausr) [inline src/assets/CloudinaryFile.ts:61]: I think there is a typo here?

**Review by strausr — APPROVED** [review]

File: src/assets/CloudinaryFile.ts (hunk 3 above)
Comment (patrick-tolosa) [reply src/assets/CloudinaryFile.ts:61]: yep, fixed.

File: src/actions/effect/vectorize.ts (hunk 2 above)
Comment (patrick-tolosa) [reply src/actions/effect/vectorize.ts:44]: added

File: src/actions/effect/vectorize.ts (hunk 1 above)
Comment (patrick-tolosa) [reply src/actions/effect/vectorize.ts:31]: added

**Pull request merged**